src/main/java/me/magnet/consultant/Consultant.java

```diff
@@ -602,8 +604,9 @@ private void updateValidatedConfig(Properties newConfig) {
 
 	/**
 	 * Tears any outstanding resources down.
+	 * @throws InterruptedException If it got interrupted while waiting for any open tasks
 	 */
-	public void shutdown() {
+	public void shutdown() throws InterruptedException {
```

Comment:
Yeah, I tried to work my way around it, but you really don't want to swallow those interrupts.